### PR TITLE
RN36727 - Updated User-defined APIs docs with CORS update

### PR DESCRIPTION
**Pull request by VanRobaysWouter**
[RN36727] [DCP213522]

### CODE REVIEW

File: user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md [hunk 1]
```diff
@@ -71,6 +71,10 @@ The errors in this section can be generated when the UserDefinableApiEndpoint Dx
 
 ## Issues when triggering user-defined APIs
 
+- **Access to XXX at 'XXX' from origin 'XXX' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.**
+
+  User-defined APIs don't set CORS headers for safety reasons. It is not safe to use a user-defined API from a web client. See [Triggering an API](xref:UD_APIs_Triggering_an_API#cors) for more info.
+
 If issues occur when you trigger a user-defined API, follow the steps below to resolve them.
```
Comment (slc-peter-vanpoucke) [inline user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md:78]: S: new item here

**Review by slc-peter-vanpoucke — APPROVED** [review]
Looks good. Only a small layout suggestion.

File: user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md (hunk 1 above)
Comment (VanRobaysWouter) [reply user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md:80]: Added subtitle

**Review by slc-peter-vanpoucke — APPROVED** [review]

File: user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md [hunk 2]
```diff
@@ -71,6 +71,12 @@ The errors in this section can be generated when the UserDefinableApiEndpoint Dx
 
 ## Issues when triggering user-defined APIs
 
+- **Access to XXX at 'XXX' from origin 'XXX' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.**
+
+  User-defined APIs don't set CORS headers for safety reasons. It is not safe to use a user-defined API from a web client. See [Triggering an API](xref:UD_APIs_Triggering_an_API#cors) for more info.
```
Comment (MariekeGO) [inline user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md:76]: Avoid contractions (e.g. "don't") in the user guide. I'd propose to rephrase this as follows:

```suggestion
  For safety reasons, user-defined APIs do not set CORS headers. It is not safe to use a user-defined API from a web client. See [Triggering an API](xref:UD_APIs_Triggering_an_API#cors).
```

File: user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Triggering_an_API.md [hunk 3]
```diff
@@ -80,6 +80,10 @@ The endpoint itself can also return [errors](#errors) with corresponding status
 
 The API will always return responses encoded in UTF-8.
 
+### CORS
+
+User-defined APIs will not return any CORS headers. It is not safe to trigger user-defined APIs from a web client using e.g. AJAX calls in JavaScript, because there is no way to safely save and use the API tokens there.
```
Comment (Skyline-ThomasGH) [inline user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Triggering_an_API.md:85]: Add a short sentence explaining the consequences. Eg. '....not return any CORS headers. It will thus not be possible to trigger a user-defined API directly from a web client.

File: user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md [hunk 4]
```diff
@@ -71,6 +71,12 @@ The errors in this section can be generated when the UserDefinableApiEndpoint Dx
 
 ## Issues when triggering user-defined APIs
 
+- **Access to XXX at 'XXX' from origin 'XXX' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.**
+
+  For safety reasons, user-defined APIs do not set CORS headers. It is not safe to use a user-defined API from a web client. See [Triggering an API](xref:UD_APIs_Triggering_an_API#cors).
```
Comment (Skyline-ThomasGH) [inline user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md:76]:
```suggestion
  For safety reasons, user-defined APIs does not set CORS headers. It is not safe to use a user-defined API from a web client. See [Triggering an API](xref:UD_APIs_Triggering_an_API#cors).
```

**Review by Skyline-ThomasGH — CHANGES_REQUESTED** [review]

**Comment by MariekeGO** [comment]
As this RN hasn't been added to a DataMiner version yet, we can't merge this in the documentation yet, so I've set this PR back to draft for now. Can you mark it as "Ready for review" when it has been merged in DataMiner?

**Comment by VanRobaysWouter** [comment]
> As this RN hasn't been added to a DataMiner version yet, we can't merge this in the documentation yet, so I've set this PR back to draft for now. Can you mark it as "Ready for review" when it has been merged in DataMiner?

@MariekeGO 
The RN doesn't change anything in behavior. CORS was configured, but it never worked. With this RN we have removed the CORS configuration from the code, but it won't change any behavior as it never worked in the first place. I marked the RN as internal as no behavior has changed. The PR can be merged because it just explains behavior that has always been in place. 
Ok if I mark it as 'ready for review' again?

**Comment by MariekeGO** [comment]
> > As this RN hasn't been added to a DataMiner version yet, we can't merge this in the documentation yet, so I've set this PR back to draft for now. Can you mark it as "Ready for review" when it has been merged in DataMiner?
> 
> @MariekeGO The RN doesn't change anything in behavior. CORS was configured, but it never worked. With this RN we have removed the CORS configuration from the code, but it won't change any behavior as it never worked in the first place. I marked the RN as internal as no behavior has changed. The PR can be merged because it just explains behavior that has always been in place. Ok if I mark it as 'ready for review' again?

Aha, that changes things, indeed! I'll just merge this PR then 😊

**Review by MariekeGO — APPROVED** [review]
